.github/workflows/cherry-pick-next-to-master.yml

```diff
@@ -1,5 +1,5 @@
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
https://github.com/marketplace/actions/github-cherry-pick-action#working-with-forked-repositories
And also https://github.com/mui/mui-x/blob/cec51c57f8e70c62e950546aee31b642d54453ca/.github/workflows/maintenance.yml#L12

Reply:
Nice, I totally missed this part 😞

Reply:
Well, all of us as well, because we approved your original PR. 🙈 🤷

Reply:
I confirmed, it worked with the new config and these settings:
<img width="831" alt="image" src="https://github.com/mui/mui-x/assets/12609561/15cdc762-7336-47ea-a5f0-eaa098dcb6d1">

Removing checkmark from `Allow Github actions to create...` makes it [fail](https://github.com/MBilalShafi/test-github-action/actions/runs/7250023602).